woocommerce/woocommerce-ios
# [Order Creation - Better Customer Selection] Retrieve a paginated list of all customers

Closes: #10306 


## Description

With this PR we synchronize with the `wc-analytics` backend to retrieve all customers through the `customers` endpoint. This is the first step to show a list of customers when creating an order so the users can select among them. 

Once we retrieve these `WCAnalyticsCustomer` objects, we convert them to `Storage.Customer` and store them. That way, we avoid having two different storage entities for customers and the Yosemite clients remain agnostic of the underlying [logic](pe5pgL-3r-p2). Even with that, the action name `synchronizeLightCustomersData` makes it clear that we don't get all customers' information after it finishes.

## Testing instructions

Make sure that your WooCommerce plugin version is 8.0.0+

As there's no UI this feature is not visible yet, but you can add this piece of code wherever it's easier for you to check that it loads the customers as expected, e.g. I added it [here](https://github.com/woocommerce/woocommerce-ios/blob/be80d6e03e4046877223f474bef59b35f3f972e4/WooCommerce/Classes/ViewRelated/Orders/Order%20Creation/EditableOrderViewModel.swift#L345) to be executed whenever we tap to create a new order:

```
stores.dispatch(CustomerAction.synchronizeLightCustomersData(siteID: siteID, pageNumber: 1, pageSize: 25, onCompletion: { result in
            let predicate = NSPredicate(format: "siteID == %lld", siteID)
            let descriptor = NSSortDescriptor(keyPath: \StorageCustomer.firstName, ascending: true)
            let resultsController = ResultsController<StorageCustomer>(storageManager: storageManager, matching: predicate, sortedBy: [descriptor])
            try? resultsController.performFetch()
            let results = resultsController.fetchedObjects

            debugPrint("customers", results.map { ($0.firstName ?? "") + " " + ($0.lastName ?? "")})
        }))
```

## Screenshots



---
